dids/src/main/kotlin/web5/sdk/dids/verificationmethods/VerificationMethod.kt

```diff
@@ -0,0 +1,125 @@
+package web5.sdk.dids.verificationmethods
+
+import com.nimbusds.jose.Algorithm
+import com.nimbusds.jose.jwk.Curve
+import com.nimbusds.jose.jwk.JWK
+import web5.sdk.crypto.KeyGenOptions
+import web5.sdk.crypto.KeyManager
+import web5.sdk.dids.PublicKey
+import web5.sdk.dids.PublicKeyPurpose
+import java.util.UUID
+
+/** Common interface for options available when adding a VerificationMethod. */
+public interface VerificationMethodSpec
+
+/** Common interface for classes that can generate a [PublicKey] from a [VerificationMethodSpec]. */
+public interface VerificationMethodGenerator {
+  /**
+   * Generates a [PublicKey] from a [VerificationMethodSpec]. The first element of the pair is an optional id that can
+   * be used to identify the private key associated with the generated public key.
+   */
+  public fun generate(): Pair<String?, PublicKey>
+}
+
+/**
+ * A [VerificationMethodSpec] where a [KeyManager] will be used to generate the underlying verification method keys.
+ * The parameters [algorithm], [curve], and [options] will be forwarded to the keyManager.
+ *
+ * [relationships] will be used to determine the verification relationships in the DID Document being created.
+ * */
+public class VerificationMethodCreationParams(
+  public val algorithm: Algorithm,
+  public val curve: Curve? = null,
+  public val options: KeyGenOptions? = null,
+  public val relationships: Iterable<PublicKeyPurpose>
+) : VerificationMethodSpec {
+  internal fun toGenerator(keyManager: KeyManager): VerificationMethodKeyManagerGenerator {
+    return VerificationMethodKeyManagerGenerator(keyManager, this)
+  }
+}
+
+private const val JsonWebKey2020Type = "JsonWebKey2020"
+
+/**
+ * A [VerificationMethodSpec] according to https://w3c-ccg.github.io/lds-jws2020/.
+ *
+ * The [id] property cannot be over 50 chars and must only use characters from the Base64URL character set.
+ */
+public class JsonWebKey2020VerificationMethod(
+  public val id: String,
+  public val controller: String? = null,
+  public val publicKeyJwk: JWK,
+  public val relationships: Iterable<PublicKeyPurpose> = emptySet()
+) : VerificationMethodSpec, VerificationMethodGenerator {
+  override fun generate(): Pair<String?, PublicKey> {
+    return Pair(null, PublicKey(id, JsonWebKey2020Type, controller, publicKeyJwk, relationships))
+  }
+}
+
+/**
+ * A [VerificationMethodSpec] according to https://w3c-ccg.github.io/lds-ecdsa-secp256k1-2019/.
+ *
+ * The [id] property cannot be over 50 chars and must only use characters from the Base64URL character set.
+ */
+public class EcdsaSecp256k1VerificationKey2019VerificationMethod(
```

Comment:
I think we should axe all usage of this in favor of JWK

Reply:
Can you elaborate on the motivation?

As a library, I think we should allow users to choose what is best fit for their purposes, as long as supporting it is not a big maintenance burden.

Reply:
in what case is it a better fit for purpose to use this instead of a jwk?

Reply:
I didn't want to guess about how other consumers would intend to use it. Thinking more about it, I can see that it's reasonable to not support it at all.

Reply:
This would result in changes to the test vectors, as some (like https://github.com/TBD54566975/web5-kt/blob/d967931830113e21ad667207feeb742e123c4458/dids/src/test/resources/publicKeyModel1.json#L2) are using this method. I'll open a separate issue to address this.

Reply:
thanks!